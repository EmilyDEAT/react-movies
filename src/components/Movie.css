.movieContainer img {
  width: 100%;
}
