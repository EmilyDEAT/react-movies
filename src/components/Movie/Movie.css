.movieImg {
  width: 100%;
  cursor: pointer;
  transition: all 300ms ease-out;
}

.movieImg:hover {
  box-shadow: .4em .4em .2em rgba(35, 37, 42, 0.3);;
  transform: scale(1.1);
}

.movieImgHidden {
  display: none;
}

.movieContainerPlaceholder {
  background-color: rgb(158, 161, 169);
  width: 100%;
  padding-top: 150%;
  animation: pulseLight 5s infinite;
}

@keyframes pulseLight {
  0% {
    opacity: 0.8;
  }

  50% {
    opacity: 0.5;
  }

  100% {
    opacity: 0.8;
  }
}