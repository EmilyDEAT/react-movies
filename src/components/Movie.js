import React from 'react'

import './Movie.css'

const Movie = ({ movie, showInfo }) => {
  return (
    <div className='movieContainer'>
      <img id={movie.id} className='movieImg' src={`https://image.tmdb.org/t/p/w500${movie.poster_path}`} alt={movie.title} onClick={showInfo}/>
    </div>
  )
}

export default Movie