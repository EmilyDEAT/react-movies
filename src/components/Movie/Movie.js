import React, { useState } from "react";

import "./Movie.css";

const Movie = ({ movie, showInfo }) => {
  const [loaded, setLoaded] = useState(false);

  const handleImgLoaded = () => {
    setLoaded(true);
  };

  return (
    <div className="movieContainer">
      {loaded ? null : <div className="movieContainerPlaceholder"></div>}
      <img
        id={movie.id}
        className={loaded ? "movieImg" : "movieImgHidden"}
        src={`https://image.tmdb.org/t/p/w500${movie.poster_path}`}
        alt={movie.title}
        onClick={showInfo}
        onLoad={handleImgLoaded}
      />
    </div>
  );
};

export default Movie;
