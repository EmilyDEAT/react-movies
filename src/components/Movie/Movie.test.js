import React from "react";
import Movie from "./Movie";
import renderer from "react-test-renderer";

it("Movie renders correctly", () => {
  const props = {
    movie: { id: 1, poster_path: "", title: "" },
    showInfo: () => {},
  };
  const movie = renderer.create(<Movie {...props} />).toJSON();
  expect(movie).toMatchSnapshot();
});
