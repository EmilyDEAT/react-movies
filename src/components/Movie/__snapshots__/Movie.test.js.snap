// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Movie renders correctly 1`] = `
<div
  className="movieContainer"
>
  <div
    className="movieContainerPlaceholder"
  />
  <img
    alt=""
    className="movieImgHidden"
    id={1}
    onClick={[Function]}
    onLoad={[Function]}
    src="https://image.tmdb.org/t/p/w500"
  />
</div>
`;
